src/teleop/gui/src/components/LimitSwitch.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+    <div class="wrap">
+      <ToggleButton
+        :id="name"
+        :current-state="limit_enabled"
+        :label-enable-text= "name + ' On'"
+        :label-disable-text= "name + ' Off'"
+        @change="toggleLimitSwitch()"
+      />
+    </div>
+</template>
+
+<script>
+    import ToggleButton from "./ToggleButton.vue";
+    import ROSLIB from "roslib/src/RosLib";
+
+    export default {
+        props: {
+            name: {
+                type: String,
+                required: true
+            },
+            state_name: {
```

Comment:
Would probably be better to call this "switch_name", state_name doesn't really make sense to me